File: Toeplitz-Matrix.java
class Solution {
    public boolean isToeplitzMatrix(int[][] m) {
        int r = m.length;
        int c = m[0].length;

        for (int i = 1; i < r; i++) 
        {
            for (int j = 1; j < c; j++) 
            {
                if (m[i][j] != m[i - 1][j - 1]) 
                {
                    return false;
                }
            }
        }
        return true;
    }
}

File: Toeplitz-Matrix.py
class Solution(object):
    def isToeplitzMatrix(self, m):
        """
        :type m: List[List[int]]
        :rtype: bool
        """
        r = len(m)
        c = len(m[0])
        for i in range(1, r):
            for j in range(1, c):
                if m[i][j] != m[i - 1][j - 1]:
                    return False
        return True
